### PR TITLE
/remind me

### PR DESCRIPTION
**Pull request by bkeepers**
@jbjonesjr this is a spike to work on what we talked about last week with slash commands.

transition_plan_for_node-github_ _issue__7_ _octokit_discussions

A few notes about this:

- The reminder logic is a hand-coded thing that uses chrono. It's working pretty well for the tests cases thrown at it so far. There are some TODOs in the tests for refiners to add (e.g. prefer work hours for times without am/pm), but could probably be rewritten to use more sophisticated NLP at some point
- I plan to extract the command code into a probot extension, so that'll get removed from this code.
- `/remind me …` is the only thing that works at the moment. Eventually it'd be cool to add `/remind @jbjonesjr …` or `/remind @probot/maintainers …`

What do you think?

### CODE REVIEW

**Comment by jbjonesjr** [comment]
couple notes:

* snooze vs remind as a verb 
  * Slack actually uses them in a weird interchangable way (when a reminder pops, you have the ability to snooze it for longer... see screenshot below). You set a reminder, you snooze it for later
  * Inbox by Google:tm: let's you "set a reminder" (aka, pop up something in the future) or "snooze a thread" (make this thread go away until some point in the future).
  * If the functionality includes auto-closing the thread and reopening it in the future, i would argue that the verb is `snooze`, but if it was just probot posting a comment at some point in the future, this is a reminder.

 Thoughts? Is remind the base case, and snooze an additional workflow built on top of it? That is starting to seem more modular to me...


----

The command itself looks good! I'm :+1: on that in general. Curious to see what the response looks like in 2 weeks


image

File: index.js [hunk 1]
```diff
@@ -1,13 +1,38 @@
+// Use UTC for for all Date parsing
+process.env.TZ = 'UTC';
```
Comment (jbjonesjr) [inline index.js:2]: that's sooo helpful.

File: index.js [hunk 2]
```diff
@@ -1,13 +1,38 @@
...
+      }
+
+      const config = await context.config('probot-snooze.yml', JSON.parse(fs.readFileSync('./etc/defaults.json', 'utf8')));
+      const freeze = new Freeze(context.github, config);
+
+      freeze.freeze(context, {
+        assignee: reminder.who,
+        unfreezeMoment: reminder.when,
+        message: reminder.what
+      });
```
Comment (jbjonesjr) [inline index.js:30]: reminder.[who|what|when] is AWESOME

**Comment by bkeepers** [comment]
> snooze vs remind as a verb…

Yeah, I agree with all of those thoughts, and I agree that these actually feel like two separate but related pieces of functionality. Are they separate apps? Or the same app with two different pieces of functionality?

> Is remind the base case, and snooze an additional workflow built on top of it? That is starting to seem more modular to me...

Snooze seems like a reminder without an explicit `who` and `what`, and with the added case of making it go away until then.

`/snooze until tomorrow` vs `/remind @jbjonesjr to merge this pull request tomorrow`. The first one should close it and reopen, the second one just pings you again tomorrow.

> Curious to see what the response looks like in 2 weeks

Here's another example. It still has the closing functionality, but based on our discussion here, it feels like it shouldn't by default. The user could always click **Close and comment** if they want it closed, and the app could still reopen it when the reminder is due.

testing_slash_commands_ _issue__109_ _robotland_test

File: index.js (hunk 1 above)
Comment (bkeepers) [reply index.js:2]: We also need to figure out what we want to do about timezones. Right now all times show up as UTC. This is mostly an issue for relative times like `tomorrow morning`.

GitHub.com uses frontend JS to get the user's timezone offset, but there's not a way for us to do that right now per-user.

Short term I was thinking we could add `timezone` to the config file as the default for the repo.

File: index.js (hunk 2 above)
Comment (bkeepers) [reply index.js:30]: 👍  I was planning to open another PR that renames the values that are stored to that.

**Pull request closed without merging**